spec/identity_spec.rb

```diff
@@ -16,24 +19,59 @@
     )
   end
 
-  before do
-    FileUtils.remove_dir(tmp_dir) if File.exists?(tmp_dir)
-    FileUtils.mkdir_p(tmp_dir)
-    File.write(File.join(tmp_dir, 'config'), identity_string)
+  before(:all) do
+    swarm_port=4061
+    api_port=5061
+    gateway_port=8090
+    @tmp_dir=ENV['TICTAC_TEST_DIR'] || File.join(__dir__, 'tmp')
```

Comment:
using attributes like this means you do a lot of grepping for the frequently used attributes.
you can get referenceable values like this by adding a let statement like in line 10.
Also your new config appears to be using whatever the user's actual private key is, the tests should use their own.

`let(:cfg) { TicTac::Config.new(tmp_dir)`
`let(:tmp_dir) { ENV['TICTAC_TEST_DIR'] || File.join(__dir__, 'tmp')`